src/student/student.service.ts

```diff
@@ -3,21 +3,96 @@ import { CreateStudentDto } from './dto/create-student.dto';
 import { UpdateStudentDto } from './dto/update-student.dto';
 import { ReadStudentDto } from './dto/read-student.dto';
 import { PrismaService } from 'src/prisma/prisma.service';
+import { createClient } from '@supabase/supabase-js';
+import { decode } from 'base64-arraybuffer';
+import { v4 as uuidv4 } from 'uuid';
 
 @Injectable()
 export class StudentService {
   constructor(private prisma: PrismaService) {}
 
-  create(createStudentDto: CreateStudentDto) {
-    const testStudent = this.prisma.testModel.create({
+  getPaymentById(createStudentDto) {
+    return this.prisma.payment.findUnique({
+      where: {
+        id: createStudentDto.paymentId,
+      },
+    });
+  }
+
+  async createPayment(payment_path: string, isRegisterByStudent: boolean) {
+    const newPayment = await this.prisma.payment.create({
+      data: {
+        photo_src: payment_path,
+        // if submission is by student, put in pending status
+        // until cashier accepts payment
+        // else submission is by cashier, accepted status
+        status: isRegisterByStudent ? 'pending' : 'accepted',
```

Comment:
okay na eto for now, but once ma setup na ang authentication, check should be done with the session variables. kung ano user type ng nag submit nung request.

Reply:
Question:
Pag naa nay OAuth, mag assume ko naa na siyay session id or unsa ba na identifier sa user type, ma send na siya sa Body sa POST method? @rhaeyx

Reply:
Yeah, pag ma setup na ang auth kay part sa session ang user type. Gi note lang nako daan basig makalimot ko ug remind pag ma implement na ang auth. Di pa ko sure asa mabutang ang data though, depende pa sa mag implement.

Reply:
Ahh noted noted. Pwede nani i mark as resolve or i wait sa nako ang pag implement sa OAuth para dili makalimtan?

Reply:
Iresolve lang, tas pa remind nalang pud or create ra ug another PR pag ma implement na ang auth.

Reply:
Resolved, thanks!
Action:
-Reminder to implement user type once OAuth is created